src/utils.c

```diff
@@ -67,7 +67,7 @@ size_t hex2bin(const char *src, void *vdst, size_t length) {
 size_t bin2hex(const void *vsrc, char *dst, size_t length) {
 	const char *src = vsrc;
 
-	for(size_t i = length; i-- > 0;) {
+	for(ssize_t i = (ssize_t) length - 1; i >= 0; --i) {
```

Comment:
I don't think this is an improvement. The original code is correct and not UB as far as I can see. Did this trigger a complaint from a sanitizer?

Reply:
UBSAN was spamming logs (about 50 reports for each test run):

    SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior net_setup.c:326:17 in
    utils.c:70:26: runtime error: unsigned integer overflow: 0 - 1 cannot be represented in type 'size_t' (aka 'unsigned long')
        #0 0x55b2263c1e88 in bin2hex tinc/src/utils.c:70:26
        #1 0x55b22630a50b in init_control tinc/src/control.c:147:2
        #2 0x55b22634eeba in setup_network tinc/src/net_setup.c:1295:6
        #3 0x55b2263c0456 in main tinc/src/tincd.c:573:6
        #4 0x7f34318690b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
        #5 0x55b2262d8ded in _start (tinc/src/tincd+0xa9ded)

I'll fix everything you noted, thanks for the review!

Reply:
Ugh. Unsigned integer overflow is *defined* behavior, at least since C99. But forcibly casting an unsigned to signed to get rid of this warning is not really nice either. Perhaps this can be replaced with:

```
for(size_t i = length; i > 0;) {
    --i;
    ...
}
```

Reply:
Oh, or should it have been `i - 1U`?